nemo/lightning/io/pl.py

```diff
@@ -155,9 +155,6 @@ def save_checkpoint(self, checkpoint: Dict[str, Any], path: _PATH, storage_optio
         checkpoint_dir = ckpt_to_weights_subdir(path, is_saving=True)
 
         fs = get_filesystem(checkpoint_dir)
-        if fs.isdir(checkpoint_dir) and dist_checkpointing.check_is_distributed_checkpoint(checkpoint_dir):
-            logging.info(f'Distributed checkpoint at path {checkpoint_dir} already exists, skipping saving')
-            return
```

Comment:
@ashors1 @maanug-nv n00b question: could you describe why this is a fix? is it unexpected behavior that we want to raise more visibly? or something else?

Reply:
> Currently on main, async save will crash if the dir exists. Non-async save will just print warning and skip saving. Looks like this change is exposing that problem in one of the tests.
> 
> With this PR, both async and synchronous checkpointing will error if we try to save to an existing directory, which I think is what we want, right? In that case, maybe we can just update the test to save to different directories

This change makes the error visible for synchronous and makes the error message more clear for both. The MCore dist_checkpointing library will raise the error now for both of these cases.

Reply:
Yes, @mikolajblaz removed this in NeMo 1 and asked us to do the same in NeMo 2. We should not attempt to save two checkpoints to the same directory, and if a user ever tries, we'd like to throw an error rather than just skipping saving.